recipes_source/bundled_inputs.rst

```diff
@@ -0,0 +1,193 @@
+    scripted_module = torch.jit.script(model)
+
+2. **Create example input and attach to model**
+
+.. code:: python
+
+    # For each method create a list of inputs and each input is a tuple of arguments
+    example_dict = {'a' : 1, 'b' : 2}
+    input = {
+        scripted_module.forward : [(torch.zeros(1,10),)],
+        scripted_module.foo : [(example_dict,)]
+    }
+
+    # Create model with bundled inputs, if type(input) is Dict then each callable key is mapped to its corresponding bundled input
+    bundled_model = bundle_inputs(scripted_module, input)
+
+
+3. **Retrieve inputs and run model on them**
+
+.. code:: python
+
+    all_info = bundled_model.get_bundled_inputs_functions_and_info()
+
+    # The return type for get_bundled_inputs_functions_and_info is complex, but essentially we are retrieving the name
+    # of a function we can use to get the bundled input for our models method
+    for func_name in all_info.keys():
+        input_func_name = all_info[func_name]['get_inputs_function_name'][0]
+        func_to_run = getattr(bundled_model, input_func_name)
+        # retrieve input
+        input = func_to_run()
+        model_function = getattr(bundled_model, func_name)
+        print(model_function(*input[0]))
+
+Inflatable args
+-------------------
+Attaching inputs to models can result in nontrivial size increases. Inflatable args are a way to compress and decompress inputs to minimize this impact.
+
+.. note:: Any automatic compression, or parsing of inflatable args only happens to top level arguments in the input tuple.
+
+   - ie if your model takes in a List type of inputs you would need to create an inflatable arg that returned a list not create a list of inflatable args.
+
+1. **Existing Inflatable args**
+
+The following input types are compressed automatically without requiring an explicit inflatable arg:
+    - Small contiguous tensors are cloned to have small storage.
+    - Inputs from torch.zeros, torch.ones, or torch.full are moved to their compact representations.
+
+.. code:: python
+
+    # bundle_randn will generate a random tensor when the model is asked for bundled inputs
+    inputs = [(torch.utils.bundled_inputs.bundle_randn((1,10)),)]
+    bundled_model = bundle_inputs(scripted_module, inputs)
+    print(bundled_model.get_all_bundled_inputs())
+
+2. **Creating your own**
+
+Inflatable args are composed of 2 parts, the deflated (compressed) argument, and an expression or function definition to inflate them.
+
+.. code:: python
+
+    def create_example(*size, dtype=None):
+        """Generate a tuple of 2 random tensors both of the specified size"""
+
+        deflated_input = (torch.zeros(1, dtype=dtype).expand(*size), torch.zeros(1, dtype=dtype).expand(*size))
+
+        # {0} is replaced with deflated[foo][bar] and is how you access your deflated value in the inflation expression
+        return InflatableArg(value=deflated_input, fmt="(torch.randn_like({0})), torch.randn_like({0}))")
```

Comment:
Q. Did you mean `return InflatableArg(value=deflated_input, fmt="(torch.randn_like({0})), torch.randn_like({1}))")` instead?